standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/StandardizationInterpreter.scala

```diff
@@ -17,109 +17,129 @@ package za.co.absa.enceladus.standardization.interpreter
       if (field.name == ErrorMessage.errorColumnName) {
-        Seq(df1.col(field.name))
+        (df.col(field.name)::accCols, accErrorCols, nextErrorColumnIndex)
       } else {
         val stdOutput = TypeParser.standardize(field, "", df.schema)
         logger.info(s"Applying standardization plan for ${field.name}")
-        stdApplyLogic(stdOutput, field)
+        val (column1, column2, errField) = stdApplyLogic(stdOutput, field, nextErrorColumnIndex)
+        (column1::column2::accCols, errField::accErrorCols, nextErrorColumnIndex + 1)
       }
-    }: _*)
+    }
 
-    // step 3 drop intermediate error columns
-    logger.info(s"Step 3.1: Preserve existing error column")
-    // preserve existing errors - rename into a temporary column
-    val errColPreserved = if (std.columns.contains(ErrorMessage.errorColumnName)) {
-      val newName = ErrorMessage.errorColumnName + (Random.nextInt.abs)
-      errorCols.append(newName)
-      std.withColumnRenamed(ErrorMessage.errorColumnName, newName)
-    } else std
+    (df.select(cols.reverse: _*), errorColNames)
+  }
 
-    logger.info(s"Step 3.2: Collect all temporary error columns")
+  private def preserveExistingErrorColumn(dataFrame: DataFrame, errorCols: ErrorCols): (DataFrame, ErrorCols) = {
+    if (dataFrame.columns.contains(ErrorMessage.errorColumnName)) {
+      // preserve existing errors - rename into a temporary column
+      val newName = s"${ErrorMessage.errorColumnName}0"
+      (dataFrame.withColumnRenamed(ErrorMessage.errorColumnName, newName), newName::errorCols)
+    } else {
+      (dataFrame, errorCols)
+    }
+  }
+
+  private def collectAllTempErrorColumns(dataFrame: DataFrame, errorCols: ErrorCols)
+                                            (implicit spark: SparkSession, udfLib: UDFLibrary): DataFrame = {
     // collect all of the error attributes into an array and flatten
     logger.info(s"Error cols: ${errorCols.toList}")
-    val collectedError = ArrayTransformations.flattenArrays(
-      errColPreserved.withColumn(ErrorMessage.errorColumnName, array(errorCols.map(col(_)): _*)),
+    ArrayTransformations.flattenArrays(
+      //reversing errorCols as err columns when collected are appended to beginning of the list
+      dataFrame.withColumn(ErrorMessage.errorColumnName, array(errorCols.reverse.map(col): _*)),
       ErrorMessage.errorColumnName)
+  }
 
-    logger.info(s"Step 3.3: Drop all temporary error columns")
-    val droppedErrs = collectedError.drop(errorCols.distinct: _*)
-    logger.info(s"Step 3.4: Clean the final error column")
-    val cleanedErr = droppedErrs.withColumn(ErrorMessage.errorColumnName, callUDF("cleanErrCol", col(ErrorMessage.errorColumnName)))
-
-    logger.info(s"Standardization process finished, returning to the application...")
-    cleanedErr
+  private def dropAllTemporaryErrorColumn(dataFrame: DataFrame, errorCols: ErrorCols): DataFrame = {
+    dataFrame.drop(errorCols.toSeq: _*)
```

Comment:
```suggestion
    dataFrame.drop(errorCols: _*)
```

It already is a list.